app/models/job.rb

```diff
@@ -1,4 +1,19 @@
 class Job < ActiveRecord::Base
   validates :title, :level_of_interest, presence: true
   belongs_to :company
+  belongs_to :category
+  has_many :comments
+
+  def show_by_count_of_jobs_by_level_of_interest
+    count_to_jobs =
+    Job.order(:level_of_interest).reverse.group_by {|job| job.level_of_interest}
+    prepared = count_to_jobs.map do |level_of_interest,job|
```

Comment:
you're doing two separate things here: you're querying the data you'd like, and you're prepping the data for output. 

the prep for the data output should live in a view helper rather in your model 👍